fix: enforce valid JSON response in interior design assistant

Anirudh2112 opened the pull request:
### Description:
**Problem:**
The interior_design_assistant app expected the model to return a valid JSON output. However, in some cases, the model's output was either invalid JSON or missing expected fields, causing runtime errors and breaking the functionality.

**Solution:**

 Implemented response_format in the turn.create calls for list_items and suggest_alternatives to enforce structured JSON outputs.

**Key Changes:**
 **Added JSON schema in response_format for both methods:**
 * list_items: Ensures output includes description (string) and items (list of strings).
 * suggest_alternatives: Ensures output is an array of objects with description (string).

yanxi0830 commented on examples/interior_design_assistant/api.py:
```diff
@@ -10,7 +10,7 @@
 from pathlib import Path
 from typing import List
-
+from examples.interior_design_assistant.utils import enforce_response_format
```
no import here?

yanxi0830 reviewed (changes requested):
You may have forgot to push changes related to `enforce_response_format`?

ashwinb commented:
This is not what the issue really is referring to. You need to use the `response_format` argument for `chat_completion()` to ensure that the underlying LLM is generating the response corresponding to the grammar you provided. Simply doing `json.loads()` and checking against your desired format is _NOT_ sufficient.

Anirudh2112 commented:
Thank you for the clarification! I’ll update the implementation accordingly and push the changes soon.

Anirudh2112 commented:
I’ve pushed the latest changes to implement `response_format` to enforce structured JSON outputs. This addresses the original issue by ensuring the LLM generates valid JSON responses directly. Let me know if there’s anything else to address!

Anirudh2112 commented:
Hi @yanxi0830, can you please go through the changes I've made and let me know if everything works as intended? Thank you.

Anirudh2112 commented:
Hi @ashwinb @yanxi0830, I've addressed all the review comments on my PRs. Just wanted to check if there are any other issues I should address or if there's anything else needed from my side to help move these forward. Thanks!
